apps/reference/docs/guides/auth/auth-helpers/nextjs-server-components.mdx

```diff
@@ -0,0 +1,712 @@
+---
+id: nextjs-server-components
+title: Supabase Auth with Next.js Server Components
+description: Authentication helpers for creating an authenticated Supabase client in Next.js 13 app directory Server Components.
+sidebar_label: 'Next.js Server Components'
+---
+
+import Tabs from '@theme/Tabs'
+
+This submodule provides experimental convenience helpers for implementing user authentication in Next.js Server Components - the `app` directory. For examples using the `pages` directory check out [Auth Helpers in Next.js](/docs/guides/auth/auth-helpers/nextjs).
+
+> To learn more about fetching and caching Supabase data with Next.js 13 Server Components, check out our [blog](/blog/fetching-and-caching-supabase-data-in-next-js-server-components) or [live stream](https://www.youtube.com/watch?v=QH0P5xZt5wY).
+
+## Install the Next.js helper library
+
+<Tabs
+  scrollable
+  size="small"
+  type="underlined"
+  defaultActiveId="npm"
+>
+
+<TabPanel id="npm" label="npm">
+
+```sh
+npm install @supabase/auth-helpers-nextjs
+```
+
+> [Next.js Server Components](https://beta.nextjs.org/docs) and the `app` directory are experimental and likely to change.
+
+</TabPanel>
+<TabPanel id="yarn" label="Yarn">
+
+```sh
+yarn add @supabase/auth-helpers-nextjs
+```
+
+> [Next.js Server Components](https://beta.nextjs.org/docs) and the `app` directory are experimental and likely to change.
+
+</TabPanel>
+</Tabs>
+
+## Set up environment variables
+
+Retrieve your project's URL and anon key from your [API settings](https://app.supabase.com/project/_/settings/api) in the dashboard, and create a `.env.local` file with the following environment variables:
+
+```bash title=".env.local"
+NEXT_PUBLIC_SUPABASE_URL=YOUR_SUPABASE_URL
+NEXT_PUBLIC_SUPABASE_ANON_KEY=YOUR_SUPABASE_ANON_KEY
+```
+
+## Creating a Supabase Client
+
+<Tabs
+  scrollable
+  size="small"
+  type="underlined"
+  defaultActiveId="js"
+>
+<TabPanel id="js" label="JavaScript">
+
+Create a new file at `/utils/supabase-browser.js` and populate with the following:
+
+```js title="/utils/supabase-browser.js"
```

Comment:
Just a suggestion but would it be better to call it `supabase-client.js` instead of `supabase-browser.js` to make it more consistent with Nextjs documentation?

Reply:
A supabase client (in Supabase terms) can be used both in browser and server environments, therefore we want to be very explicit here.